mod/01_rating/spec/set/right/downvotee_search_spec.rb

```diff
@@ -121,7 +121,16 @@
               "+company"=>{"content"=>"[[#{company.name}]]",:type_id=>Card::PointerID},
               "+value"=>{"content"=>"Na einai kalytero anthropo apo ton patera toy", :type_id=>Card::PhraseID},
               "+year"=>{"content"=>"2015", :type_id=>Card::PointerID},
-              "+Link"=>{:content=>"http://www.google.com/?q=fringepeter", "type_id"=>Card::PhraseID}
+              "+source"=>{
+                "subcards"=>{
+                  "new source"=>{
+                    "+Link"=>{
+                      "content"=>"http://www.google.com/?q=fringepeter",
+                       "type_id"=>Card::PhraseID
+                    }
+                  }
+                }
+              }
```

Comment:
Very thorough answer, Henry. Thank you.

I guess I was looking at old code on the +Link issue.

It seems to me like the need to add a fake name ("new source") for the API to get things to work is a wart.  We probably need some standard way to handle autonamed cards. (eg *autoname_1).  But the "new source" workaround makes sense for now. (In general, it's best to use a "*" in front of temporary names just to reduce the chance of conflict).

If I understand correctly, the added complexity here is really about the fact that our API can now handle trees.  Before, if we had a form within a form within a form, we would effectively compress them all so that all the cards were subcards of one parent.  Now there can be many generations.  This is really powerful, because it means, for example, that you could create a setup where the +Link is required for a Source but a source is not required for a Note (or similar custom rules).

Thanks for clarifying.  I'm going to merge and deploy this first so that it's easier to review and test the other PR.